sig-gating-contracts/test/NexeraIDSignerManager.spec.ts

```diff
@@ -70,28 +70,56 @@ describe(`NexeraIDSignerManager`, function () {
       txAuthInput
     );
 
-    // try to mint nft
-    const tx = await exampleGatedNFTMinter
-      .connect(testerSigner)
-      .mintNFTGated(
-        recipient,
-        signatureResponse.blockExpiration,
-        signatureResponse.signature
-      );
+    // Encoding the blockExpiration (uint256) and signature (bytes)
+    const abiEncodedBlockExpiration = ethers.utils.hexZeroPad(
+      ethers.BigNumber.from(signatureResponse.blockExpiration).toHexString(),
+      32
+    );
+
+    const length = ethers.utils.hexZeroPad(
+      ethers.BigNumber.from(96).toHexString(),
+      32
+    );
+
+    const lengthSig = ethers.utils.hexZeroPad(
+      ethers.BigNumber.from(65).toHexString(),
+      32
+    );
+
+    const unsignedTx =
+      await exampleGatedNFTMinter.populateTransaction.mintNFTGated(recipient);
+
+    const finalSig_ =
+      lengthSig + signatureResponse.signature.slice(2) + "0".repeat(62);
+
+    // Complete data
+    const txData =
+      unsignedTx.data +
+      abiEncodedBlockExpiration.slice(2) +
+      length.slice(2) +
+      finalSig_.slice(2);
```

Comment:
I don't really understand the whole process of building that txData. Since we are doing it all over the tests and we will also use it in the examples in monorepo and examples repo, maybe it would be better to refactor this in a function in the sdk and export it (contracts-sdk/lib).
And then maybe we could add comments there to explain a bit what is going on.
Because it is a lot of code for our customers to add in their front end.

Reply:
We can do this in another PR

Reply:
Well, we changed the data sent so I think we don't need this now

Reply:
I think we can resolve this because we've changed to packed encoding of `blockExpiration` and `signature`